### PR TITLE
Update restraint documentation under "Algorithms"

### PR DESCRIPTION
**Pull request by jchodera**
This attempts to fill in some details about how YANK handles restraints.

### CODE REVIEW

File: docs/algorithms.rst [hunk 1]
```diff
@@ -154,10 +155,99 @@ To correct for this, we utilize the **anisotropic long-range dispersion correcti
...
+Spherically symmetric restraints
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Harmonic resstraints (``Harmonic``)
+"""""""""""""""""""""""""""""""""""
+
+A harmonic potential is imposed between the closest atom to the center of the receptor and the closest atom to the center of the ligand, given the initial geometry.
+The equilibrium distance is zero, while the spring constant is selected such the potential reaches ``kT`` at one radius of gyration.
+This allows the ligand to explore multiple binding sites---including internal sites---without drifting away from the receptor.
+For implicit and explicit solvent calculations, harmonic restraints should be imposed and at full strength and retained throughout all alchemical states to define the bound complex.
```
Comment (Lnaden) [inline docs/algorithms.rst]: "should be imposed at full strength and retained..."

File: docs/algorithms.rst [hunk 2]
```diff
@@ -154,10 +155,99 @@ To correct for this, we utilize the **anisotropic long-range dispersion correcti
...
+
+Flat-bottom restraints (``FlatBottom``)
+"""""""""""""""""""""""""""""""""""""""
+
+A variant of ``Harmonic`` where the restraint potential is zero in the central region and grows as a half-harmonic potential outside of this region.
+A lengthscale ``sigma`` is computed from the median absolute distance from the central receptor atom to all atoms, multiplied by 1.4826.
+The transition from flat to harmonic occurs at ``r0 = 2*sigma + 5*angstroms``.
+A spring constant of ``K = 0.6 * kilocalories_per_mole / angstroms**2`` is used.
+This restraint is described in detail in :cite:`Shirts2013:yank`.
+For implicit and explicit solvent calculations, flat-bottom restraints should be imposed and at full strength and retained throughout all alchemical states to define the bound complex.
```
Comment (Lnaden) [inline docs/algorithms.rst]: "imposed at full strength..."

File: docs/algorithms.rst [hunk 3]
```diff
@@ -154,10 +155,99 @@ To correct for this, we utilize the **anisotropic long-range dispersion correcti
...
+Orientational restraints are used to confine the ligand to a single binding pose.
+
+.. warning:: Because the ligand is highly restrained orientationally, the initial configuration should have the ligand well-placed in the binding site; errors in initial pose cannot be easily recovered from.
+
+Boresch restraints (``Boresch``)
+""""""""""""""""""""""""""""""""
+
+A common type of **orientational restraints** between receptor and ligand :cite:`Boresch2003`.
+These restrain a distance, two angles, and three torsions in an attempt to keep the ligand in a specific relative binding pose.
+Default spring constants used in [1] are used, and a set of atoms is automatically chosen (three each in the ligand and receptor) to ensure that the distance (``r_aA0``) is within [1,4] Angstroms and the angles (``theta_A0``, ``theta_B0``) is several standard deviations away from 0 and ``pi``.
```
Comment (Lnaden) [inline docs/algorithms.rst]: Broken citation?

**Comment by jchodera** [comment]
Thanks!

I totally agree that lots of other sections of this page need updating. Will see how much I can get through tonight

File: docs/algorithms.rst [hunk 4]
```diff
@@ -20,7 +25,8 @@ YANK provides the somewhat unique ability to perform alchemical free energy calc
 Solvent model
 ^^^^^^^^^^^^^
 
-For the ``yank prepare amber`` command that imports AMBER ``prmtop`` files, any of the AMBER implicit solvent models available in OpenMM are available for use via the :code:`--gbsa <model>`` argument:
+For the ``yank prepare amber`` command that imports AMBER ``prmtop`` files, any of the AMBER implicit solvent models
```
Comment (jchodera) [inline docs/algorithms.rst]: I think we want to ditch the `yank prepare amber` stuff totally. It will just cause confusion.

**Comment by Lnaden** [comment]
I removed all references to the `prepare` cli. @jchodera can you double check the HREX w/ Gibbs sampling section to make sure I have that right?

**Comment by jchodera** [comment]
Looks good for now. We'll continue to iteratively refine this documentation as we progress toward 1.0 and the paper.

**Pull request merged**